src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CAConfiguration.java

```diff
@@ -58,18 +62,36 @@ private CAConfiguration(List<String> caTypes, CertificateStore.CAType caType,
      * Factory method for creating an immutable CAConfiguration from the service configuration.
      *
      * @param configurationTopics the configuration key of the service configuration
-     *
-     * @throws URISyntaxException if invalid certificateUri or privateKeyUri provided.
      */
-    public static CAConfiguration from(Topics configurationTopics) throws URISyntaxException {
+    public static Result<CAConfiguration> from(Topics configurationTopics) {
```

Comment:
With this changes we are being more specific about what went wrong in the configuration and reflect what is a valid configuration.

Reply:
Hmmm.. this seems unnecessarily complicated. How does it convey more specific information back to the caller?
I'd prefer just returning a `CAConfiguration` or throwing here. This is kind of a weird pattern to have a factory method return a Result.